src/components/FilterDrawer.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { FunctionComponent } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import SwipeableDrawer from "@material-ui/core/SwipeableDrawer";
+import List from "@material-ui/core/List";
+import TextField from "@material-ui/core/TextField";
+import FilterList from "./FilterList";
+import Button from "@material-ui/core/Button";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+
+const useStyles = makeStyles({
+  list: {
+    width: 250,
+  },
+  fullList: {
+    width: "auto",
+  },
+});
+
+interface FilterDrawerProps {
+  open: boolean;
+  onClose: () => void;
+  onOpen: () => void;
+  closeDrawer: () => void;
+  filters: { [k: string]: boolean };
+  visibleFilters: Set<string>;
+  searchString: string;
+  setSearchString: React.Dispatch<React.SetStateAction<string>>;
+  matchAny: boolean;
+  setMatchAny: React.Dispatch<React.SetStateAction<boolean>>;
+  setFieldValue: (field: string, value: number | string | boolean) => void;
+}
+const FilterDrawer: FunctionComponent<FilterDrawerProps> = ({
+  searchString,
+  setSearchString,
+  filters,
+  visibleFilters,
+  open,
+  onClose,
+  onOpen,
+  matchAny,
+  setMatchAny,
+  closeDrawer,
+  setFieldValue,
+}) => {
+  const classes = useStyles();
+  // Still need to add X to clear textbox and close drawer on enter
+
+  // Set anchor for any/all match menu
+  // handleClick of any/all match menu button
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>): void => {
+    setAnchorEl(event.currentTarget);
+  };
+  // handleClose of any/all match menu
+  const handleClose = (): void => {
+    setAnchorEl(null);
+  };
+
+  return (
+    <SwipeableDrawer
+      open={open}
+      anchor="right"
+      onClose={onClose}
+      onOpen={onOpen}
+    >
+      <div className={classes.list} role="presentation">
+        <List
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            justifyContent: "center",
+            textAlign: "center",
+          }}
```

Comment:
I see there are quite a few inline styles in this file.
I've committed some of these as well, but if possible try to bring in `useStyles` and, I know we haven't done it yet, but those `useStyles` functions can be extracted in files for reuse across the codebase.